### PR TITLE
Improve repl

### PR DESCRIPTION
**Pull request by Redempt**
Uses the `rustyline` crate to read lines, giving a much nicer prompt

### CODE REVIEW

**Comment by Redempt** [comment]
why

File: src/interpreter/repl.rs [hunk 1]
```diff
@@ -20,50 +23,90 @@ impl<'a> Default for FenderRepl<'a> {
...
+                }
+                Err(ReadlineError::Interrupted) => self.buffer.clear(),
+                Err(ReadlineError::Io(e)) => eprintln!("I/O error: {e}"),
+                Err(ReadlineError::WindowResized) => continue,
+                #[cfg(target_os = "windows")]
+                Err(ReadlineError::Decode(e)) => eprintln!("Could not decode input: {e}"),
+                #[cfg(target_os = "windows")]
+                Err(ReadlineError::SystemError(e)) => eprintln!("System error: {e}"),
+                Err(ReadlineError::Eof) => {
+                    println!("\nGoodbye");
```
Comment (FuzzyNovaGoblin) [inline src/interpreter/repl.rs:105]: remove this

File: src/interpreter/repl.rs [hunk 2]
```diff
@@ -20,50 +23,90 @@ impl<'a> Default for FenderRepl<'a> {
...
-            parent: None,
-            return_target: engine.create_return_target(),
-            num_stack_vars: 0,
-        };
-        FenderRepl { engine, scope }
+        let scope = LexicalScope::new(ArgCount::Fixed(0), engine.create_return_target());
+        FenderRepl {
+            engine,
+            scope,
+            editor: DefaultEditor::new().expect("Prompt created"),
```
Comment (FuzzyNovaGoblin) [inline src/interpreter/repl.rs:41]: this sounds like the error msg is saying what happened is actually what failed to happen

**Review by FuzzyNovaGoblin — CHANGES_REQUESTED** [review]
is there no way to edit the prior line if it was unbalanced with rustyline?

**Comment by Redempt** [comment]
I think there's a way but I haven't looked into it yet, I think I'd have to do that with the Helper type in rustyline

**Comment by Redempt** [comment]
Found it, https://docs.rs/rustyline/latest/rustyline/validate/trait.Validator.html will do that later

**Review by FuzzyNovaGoblin — APPROVED** [review]

File: src/interpreter/repl.rs [hunk 3]
```diff
@@ -20,50 +22,80 @@ impl<'a> Default for FenderRepl<'a> {
...
-                Ok(FenderReference::FRaw(FenderValue::Null)) => (),
-                Ok(val) => println!("{}", val.to_string()),
-                Err(err) => eprintln!("{err:#}"),
+        loop {
+            let line = self.editor.read_line(&FenderPrompt);
+            match line {
+                Ok(Signal::Success(line)) => {
+                    display_result(self.run_statement(line.trim()));
+                }
+                Ok(Signal::CtrlC) => continue,
```
Comment (FuzzyNovaGoblin) [inline src/interpreter/repl.rs:95]: if there is nothing on the line this should kill the program

**Review by FuzzyNovaGoblin — CHANGES_REQUESTED** [review]

File: src/interpreter/repl.rs (hunk 3 above)
Comment (Redempt) [reply src/interpreter/repl.rs:95]: How do we know if the line was empty when they hit Ctrl + C? It doesn't give us the line in that case

**Review by FuzzyNovaGoblin — APPROVED** [review]

**Pull request merged**